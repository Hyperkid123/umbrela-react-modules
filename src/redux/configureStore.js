import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
