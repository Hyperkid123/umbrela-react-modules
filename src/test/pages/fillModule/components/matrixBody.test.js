import React from 'react'
import MatrixBody, {MatrixBody as Snapshot} from '../../../../pages/fillModule/components/matrixBody';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Matrix body component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.answers = {};
    initialProps.answerMatrixMulti = jest.fn();
    initialProps.answerMatrixSingle = jest.fn();
    initialProps.questionId = 0;
    initialProps.options = [];
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  const mockStore = configureStore();
  let initialState = {};
  let store;
  beforeEach(() => {
    initialState = {
      answers: {},
      locale: {
        languages: [{
          name: "English",
          code: 'en',
          active: true,
        }],
        translations: {}
      }
    }
  });

  it('should create component correctly', () => {
    store = mockStore(initialState);
    let wrapper = shallow(
      <MatrixBody
        questionId={0}
        store={store}
        options={[]}
      />);
    expect(wrapper).toBeDefined();
  });

  it('should call single answer action', () => {
    const answerMatrixSingle = jest.fn();
    const wrapper = shallow(
      <Snapshot {...initialProps} answerMatrixSingle={answerMatrixSingle}/>
    )
    wrapper.instance().handleMatrixSingleAnswer(1, 10, false)
    expect(answerMatrixSingle).toHaveBeenCalledWith(0, 1, 10, false)
  })

  it('should call multi answer action', () => {
    const answerMatrixMulti = jest.fn();
    const wrapper = shallow(
      <Snapshot {...initialProps} answerMatrixMulti={answerMatrixMulti}/>
    )
    wrapper.instance().handleMatrixMultiAnswer(1, 10, false)
    expect(answerMatrixMulti).toHaveBeenCalledWith(0, 1, 10, false)
  })
  let options = [];
  beforeEach(() => {
    options = [{
      optionId: 0,
      optionType: 'ColumnOption',
      optionTitle: 'option column 0'
    }, {
      optionId: 1,
      optionType: 'ColumnOption',
      optionTitle: 'option column 1'
    }, {
      optionId: 2,
      optionType: 'RowOption',
      optionTitle: 'option row 0'
    }]
  });

  it('should render correctly with options', () => {
    const tree = shallow(
      <Snapshot {...initialProps} options={options}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render correctly with row options as images', () => {
    const tree = shallow(
      <Snapshot
        {...initialProps}
        options={options}
        questionType='MatrixSingleImageQuestion'
      />
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render correctly with checkboxes', () => {
    const tree = shallow(
      <Snapshot
        {...initialProps}
        options={options}
        questionType='MatrixMultiQuestion'
      />
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should render correctly with radiobuttons', () => {
    const tree = shallow(
      <Snapshot
        {...initialProps}
        options={options}
        questionType='MatrixSingleQuestion'
      />
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  it('should update state on image preview open', () => {
    const wrapper = shallow(
      <Snapshot {...initialProps}/>
    )
    wrapper.instance().handleOpen('image.foo');
    expect(wrapper.state()).toEqual({
      modalOpen: true,
      previewImage: 'image.foo'
    })
  })

  it('should update state on image preview close', () => {
    const wrapper = shallow(
      <Snapshot {...initialProps}/>
    )
    wrapper.instance().handleClose();
    expect(wrapper.state().modalOpen).toBeFalsy()
  })

  it('should update state on new props', () => {
    const wrapper = shallow(
      <Snapshot {...initialProps}/>
    )
    wrapper.setProps({});
    expect(wrapper.state().modalOpen).toBeFalsy()
  })

  it('should return false answer value if no answers exist', () => {
    const wrapper = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(wrapper.instance().getChecked(0, 0)).toBeFalsy();
  })

  it('should return correct answer value if answers exist', () => {
    const answer = {
      "33": [10, 11, 5]
    }
    const wrapper = shallow(
      <Snapshot {...initialProps} answer={answer}/>
    )
    expect(wrapper.instance().getChecked(0, 0)).toBeFalsy();
    expect(wrapper.instance().getChecked(33, 11)).toBeTruthy();
    expect(wrapper.instance().getChecked(33, -10)).toBeFalsy();
  })
});
