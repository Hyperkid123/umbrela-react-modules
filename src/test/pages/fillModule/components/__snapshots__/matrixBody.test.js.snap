// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matrix body component should render correctly empty 1`] = `
<div>
  <styled.table>
    <styled.thead>
      <styled.tr>
        <styled.td
          key="question"
        />
      </styled.tr>
    </styled.thead>
    <styled.tbody />
  </styled.table>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogContent)>
      <WithStyles(DialogTitle)>
        Náhled
      </WithStyles(DialogTitle)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`Matrix body component should render correctly with checkboxes 1`] = `
<div>
  <styled.table>
    <styled.thead>
      <styled.tr>
        <styled.td
          key="question"
        />
        <styled.td
          key="0"
        >
          option column 0
        </styled.td>
        <styled.td
          key="1"
        >
          option column 1
        </styled.td>
      </styled.tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        filler={true}
        key="2"
      >
        <styled.td>
          <span>
            option row 0
          </span>
        </styled.td>
        <styled.td
          key="0"
        >
          <WithStyles(Checkbox)
            checked={false}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
        <styled.td
          key="1"
        >
          <WithStyles(Checkbox)
            checked={false}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogContent)>
      <WithStyles(DialogTitle)>
        Náhled
      </WithStyles(DialogTitle)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`Matrix body component should render correctly with options 1`] = `
<div>
  <styled.table>
    <styled.thead>
      <styled.tr>
        <styled.td
          key="question"
        />
        <styled.td
          key="0"
        >
          option column 0
        </styled.td>
        <styled.td
          key="1"
        >
          option column 1
        </styled.td>
      </styled.tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        filler={true}
        key="2"
      >
        <styled.td>
          <span>
            option row 0
          </span>
        </styled.td>
        <styled.td
          key="0"
        >
          <WithStyles(Checkbox)
            checked={false}
            checkedIcon={<pure(RadioButtonChecked) />}
            icon={<pure(RadioButtonUnchecked) />}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
        <styled.td
          key="1"
        >
          <WithStyles(Checkbox)
            checked={false}
            checkedIcon={<pure(RadioButtonChecked) />}
            icon={<pure(RadioButtonUnchecked) />}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogContent)>
      <WithStyles(DialogTitle)>
        Náhled
      </WithStyles(DialogTitle)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`Matrix body component should render correctly with radiobuttons 1`] = `
<div>
  <styled.table>
    <styled.thead>
      <styled.tr>
        <styled.td
          key="question"
        />
        <styled.td
          key="0"
        >
          option column 0
        </styled.td>
        <styled.td
          key="1"
        >
          option column 1
        </styled.td>
      </styled.tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        filler={true}
        key="2"
      >
        <styled.td>
          <span>
            option row 0
          </span>
        </styled.td>
        <styled.td
          key="0"
        >
          <WithStyles(Checkbox)
            checked={false}
            checkedIcon={<pure(RadioButtonChecked) />}
            icon={<pure(RadioButtonUnchecked) />}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
        <styled.td
          key="1"
        >
          <WithStyles(Checkbox)
            checked={false}
            checkedIcon={<pure(RadioButtonChecked) />}
            icon={<pure(RadioButtonUnchecked) />}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogContent)>
      <WithStyles(DialogTitle)>
        Náhled
      </WithStyles(DialogTitle)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;

exports[`Matrix body component should render correctly with row options as images 1`] = `
<div>
  <styled.table>
    <styled.thead>
      <styled.tr>
        <styled.td
          key="question"
        />
        <styled.td
          key="0"
        >
          option column 0
        </styled.td>
        <styled.td
          key="1"
        >
          option column 1
        </styled.td>
      </styled.tr>
    </styled.thead>
    <styled.tbody>
      <styled.tr
        filler={true}
        key="2"
      >
        <styled.td>
          <div>
            <styled.img
              onClick={[Function]}
              src="option row 0"
            />
          </div>
        </styled.td>
        <styled.td
          key="0"
        >
          <WithStyles(Checkbox)
            checked={false}
            checkedIcon={<pure(RadioButtonChecked) />}
            icon={<pure(RadioButtonUnchecked) />}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
        <styled.td
          key="1"
        >
          <WithStyles(Checkbox)
            checked={false}
            checkedIcon={<pure(RadioButtonChecked) />}
            icon={<pure(RadioButtonUnchecked) />}
            onChange={[Function]}
            style={
              Object {
                "borderLeft": "16px solid transparent",
                "margin": "auto",
                "width": 40,
              }
            }
          />
        </styled.td>
      </styled.tr>
    </styled.tbody>
  </styled.table>
  <WithStyles(Dialog)
    maxWidth={false}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(DialogContent)>
      <WithStyles(DialogTitle)>
        Náhled
      </WithStyles(DialogTitle)>
      <LazyLoad
        offset={0}
        once={false}
        overflow={false}
        resize={false}
        scroll={true}
        unmountIfInvisible={false}
      >
        <styled.img
          alt=""
          src=""
        />
      </LazyLoad>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        raised={true}
      >
        Zavřít
      </WithStyles(Button)>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;
